pandora-client-web/src/components/characterRestrictionOverride/characterRestrictionOverride.tsx

```diff
@@ -0,0 +1,213 @@
+	const roomContext = useChatroomRequired();
+	const state = useCharacterState(roomContext, player.id);
+	const restrictionOverride = state?.restrictionOverride;
+
+	useKeyDownEvent(useCallback(() => {
+		hide();
+		return true;
+	}, [hide]), 'Escape');
+
+	const [doModeExit, exiting] = useAppearanceActionEvent({
+		type: 'restrictionOverrideChange',
+		mode: 'normal',
+	});
+	const [doSafeModeEnter, enteringSafeMode] = useAppearanceActionEvent({
+		type: 'restrictionOverrideChange',
+		mode: 'safemode',
+	});
+	const [doTimeoutModeEnter, enteringTimeoutMode] = useAppearanceActionEvent({
+		type: 'restrictionOverrideChange',
+		mode: 'timeout',
+	});
+	const processing = exiting || enteringSafeMode || enteringTimeoutMode;
+
+	return (
+		<ModalDialog>
+			<h3>
+				Safemode
+			</h3>
+			<CharacterSafemodeHelpText />
+			<p>
+				Safemode should be seen as a last resort for when OOC-talk about limits fails, or you<br />
+				are not fine with what is happening and require a quick out to feel safe again. Please be mindful that<br />
+				there is a real person behind every character. Therefore, it is recommended to communicate issues first<br />
+				Out of Character (OOC) by using the '/ooc' command or by prefixing a message with<br />
+				double round brackets '(('.
+			</p>
+			<CharacterTimeoutModeHelpText />
+			{
+				restrictionOverride != null ? (
+					<CharacterRestrictionOverrideLeave
+						{ ...restrictionOverride }
+						doModeExit={ doModeExit }
+						processing={ processing }
+					/>
+				) : (
+					<>
+						<p>
+							<i>You are currently not in safemode or timeout mode.</i>
+						</p>
+						<p>
+							<strong>Warning:</strong> After entering safemode, you will not be able to leave it for { FormatTimeInterval(GetRestrictionOverrideConfig('safemode').allowLeaveAt) }!<br />
+							(Timeout mode can toggled on off without any time restrictions.)
+						</p>
+						<Row padding='medium' alignX='space-between'>
+							<Button onClick={ hide }>Cancel</Button>
+							<Button onClick={ doTimeoutModeEnter } disabled={ processing }>
+								Enter timeout mode!
+							</Button>
+							<Button onClick={ doSafeModeEnter } disabled={ processing }>
+								Enter safemode!
+							</Button>
+						</Row>
+					</>
+				)
+			}
+		</ModalDialog>
+	);
+}
+
+export function GetRestrictionOverrideText(type: RestrictionOverride['type']): string {
+	switch (type) {
+		case 'safemode':
+			return 'safemode';
+		case 'timeout':
+			return 'timeout mode';
+	}
+}
+
+function CharacterRestrictionOverrideLeave({ type, allowLeaveAt, doModeExit, processing }: RestrictionOverride & {
+	doModeExit: () => void;
+	processing: boolean;
+}): ReactElement {
+	const currentTime = useCurrentTime();
+	const canLeave = currentTime >= allowLeaveAt;
+	const mode = GetRestrictionOverrideText(type);
+	const { hide } = useRestrictionOverrideDialogContext();
+
+	return (
+		<>
+			<p>
+				<strong>You are currently in a { mode }!</strong><br />
```

Comment:
I would like to remove the "a" as it is not unspecific. There is just one timeout mode for your character.